### PR TITLE
Add support for Basic authentication on SonarScanner mirrors

### PR DESCRIPTION
**Pull request by victor-diez-sonarsource**
Cherry-picked from #44

### CODE REVIEW

**Comment by interone-ms** [comment]
@victor-diez-sonarsource regarding your documentation change, the password protection in this PR is for the mirror itself (i.e. the Nexus instance), *not* the proxy - proxy auth is separate, as part of the proxy URL.

**Comment by victor-diez-sonarsource** [comment]
Yes, I understand. But you mentioned:

```
Mirror authentication is supported as well, see below.
```

but there was nothing after that. So I understood you were referring to the authentication for proxies, which was already documented in the first PR.

File: src/config.js [hunk 1]
```diff
@@ -104,9 +106,14 @@ function getExecutableParams(params = {}) {
...
   if (proxy && proxy !== '') {
     const proxyAgent = new HttpsProxyAgent(proxy);
-    config.httpOptions = {
-      httpRequestOptions: { agent: proxyAgent },
-      httpsRequestOptions: { agent: proxyAgent },
+    config.httpOptions.httpRequestOptions = { agent: proxyAgent };
+    config.httpOptions.httpsRequestOptions = { agent: proxyAgent };
+  }
+
+  if (finalUrl.username !== '' || finalUrl.password !== '') {
```
Comment (yassin-kammoun-sonarsource) [inline src/config.js:113]: What would happen if one of them is missing? Should the two of them be required?

Comment (yassin-kammoun-sonarsource) [reply src/config.js:113]: 👍

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM

**Comment by victor-diez-sonarsource** [comment]
@interone-ms Please let me know if it's all clear or you meant to say something else in the README. Would like to merge this before next release.

**Comment by interone-ms** [comment]
@victor-diez-sonarsource you're right, good catch. Looks like I reordered stuff during writing and didn't update the wording.

**Pull request merged**